src/operator/numpy/np_polynomial_op.cc

```diff
@@ -53,7 +53,8 @@ struct polyval_backward_p {
     DType igrad_p = 0;
     index_t j = x_size - 1;
     while (j >= 0) {
-        igrad_p += pow(x_dptr[j], p_size - i - 1) * ograd_dptr[j];
+        igrad_p += pow(x_dptr[j], static_cast<DType>(p_size) -
```

Comment:
just to confirm, there is a unittest coverage for this change?

Reply:
yes: https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_numpy_op.py#L9394